hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -367,6 +367,7 @@ public void revertChildrenOf(@NonNull final SingleTransactionRecordBuilderImpl r
                 childRecordBuilders.set(i, null);
             } else {
                 if (child.reversingBehavior() == ReversingBehavior.REVERSIBLE && SUCCESSES.contains(child.status())) {
+                    child.tokenTransferLists().clear();
```

Comment:
Awesome catch. 🙌 I created [this issue](https://github.com/hashgraph/hedera-services/issues/10308) to give `SingleTransactionRecordBuilderImpl` a method to null out all its "side-effect" fields like `tokenTransferLists`.